src/plugin/memory-provider.ts

```diff
@@ -33,12 +33,14 @@ export class MemoryProvider {
     public static WriteKey = `${manifest.PACKAGE_NAME}.canWrite`;
 
     protected readonly sessions = new Map<string, DebugProtocol.Capabilities | undefined>();
-    protected adapterRegistry?: AdapterRegistry;
 
-    public activate(context: vscode.ExtensionContext, registry: AdapterRegistry): void {
-        this.adapterRegistry = registry;
+    constructor(protected adapterRegistry: AdapterRegistry) {
+
+    }
+
+    public activate(context: vscode.ExtensionContext): void {
```

Comment:
I think we went back and forth on a previous PR re: constructors vs. activate. I'm not sure I'm a fan of having both.

Reply:
This was the only class which wasn't aligned to the existing pattern of:

- create new thing
- activate thing

In the extension entrypoints.

Construction and activation are distinct actions I see as:

- Construction composes classes together
- Activation loads the functionality into the context. It may be async (constructors can't do that) and its possible it could get deactivated and reactivated in future

Therefore I personally prefer to keep them separate and not activate during construction nor pass composition classes in to activation functions.